module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -130,13 +130,100 @@ public function addCss($css)
     public function addJs($js)
     {
         if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+            $this->addJsEntry($js);
+        } elseif (isset($js['file'])) {
+            $this->addJsEntry($js);
+        } elseif (isset($js[0])) {
+            foreach ($js as $current) {
+                $this->addJsEntry($current);
+            }
+        } else {
+            trigger_error("Invalid JS entry format: " . print_r($js, true));
+        }
+    }
+
+    /**
+     * Helper function for adding a Javascript file.
+     *
+     * @param string|array $jsEntry Entry to add, either as string with path
+     * or array with additional properties.
+     *
+     * @return void
+     */
+    protected function addJsEntry($jsEntry)
+    {
+        if (!is_array($jsEntry)) {
+            $parts = $this->parseSetting($jsEntry);
+            if (count($parts) == 1) {
+                $jsEntry = ['file' => $jsEntry];
+            } else {
+                $jsEntry = [
+                    'file' => $parts[0],
+                    'attributes' => ['conditional' => trim($parts[1])],
+                ];
+            }
         }
-        foreach ($js as $current) {
-            $this->js[] = $current;
+        if (!isset($jsEntry['position'])) {
+            $jsEntry['position'] = 'header';
+        }
+
+        if (!in_array($jsEntry, $this->js, true)) {
```

Comment:
Just food for thought, but do we want to test for duplicates only using full array matches, or would it be better to match on a subset of attributes, so that a sub-theme can override a parent theme's settings? Or perhaps an easier approach would be to consider an `override` boolean attribute that could cause a child theme's settings for a file to replace those already loaded from a parent theme. Not something we need to actually implement before there's a real use case, but just a possibility to consider while designing this.

Reply:
You're right! If we allow to override an existing dependency, we should be aware of the consequences. For example => If the overridden settings contain a different priority or dependency, they need to get moved. This also means that all other entries that are depending on it need to be re-evaluated.
If we don't need this right now and the recursion is not implemted, would it be better to throw an Exception in this case (for consistency reasons)?

Reply:
I agree that it's probably not worth implementing more complex logic here unless we have a use case. I'm fine with throwing the exception as you have implemented, but we might also consider simply ignoring duplicates and logging a warning. (For now, I think the exception is probably a better idea, since it's less ambiguous and forces the user to fix the problem... but I wanted to mention the alternative for the sake of discussion; it might be preferable if we think there are likely to be real-world scenarios where custom themes already have duplicate filenames, since it would be more forgiving during upgrades... but fortunately, I don't think that scenario is actually very likely at all).